ospf6d/ospf6_lsdb.c

```diff
@@ -92,6 +106,52 @@ static void _lsdb_count_assert(struct ospf6_lsdb *lsdb)
 #define ospf6_lsdb_count_assert(t) ((void) 0)
 #endif /*DEBUG*/
 
+#define OSPF6_LSDB_STATS_NUM_INC 1
+#define OSPF6_LSDB_STATS_NUM_DEC 0
+
+/* Increment or decrement LSA-Type Count in Database LSA Statistics */
+static inline void ospf6_lsdb_stats_num_incdec(struct ospf6_lsa *lsa,
+					       struct ospf6_lsdb *lsdb,
+					       uint8_t incdec)
```

Comment:
Can't we just have `int8_t count` and do `(*num) += count` at the end of this function?
Why do we need some special definitions to simply do +1/-1 on a number?

Reply:
This function was written to be used to either increment or decrement a counter in a specific data structure by one at addition or removal of an LSA to or from the link state database. There is absolutely no other intended use case.

The special definitions are supposed to make the code more readable. Any C (pre-)compiler should optimize these constants the same way as if i would use `0` or `1` constant integers in the inline function call.

Does this make sense?